web/src/components/Navigation/TopNav.tsx

```diff
@@ -178,17 +181,22 @@ export default function TopNav({ searchValue, setSearchValue }: Props) {
               <PopoverButton
                 button={IconButton}
                 buttonChildren={<MdSettings className={"h-full w-10"} />}
+                popoverPlacement={"bottom-end"}
                 rotateButtonOnOpen={true}
               >
                 {({ close }) => (
                   <div className={"flex flex-col gap-1"}>
                     <Button
                       variant={"text"}
-                      onClick={() => {}}
+                      onClick={() => {
+                        changeCurrentTheme(
+                          currentTheme === "light" ? "dark" : "light"
```

Comment:
I initially had it like that but I removed it at some point because defaulting it to light would contribute to the dom content flashing issue. Also, in this case I dont think it would help though, since we are basically setting it to the opposite of the currentTheme regardless

Reply:
Oh, you're right about the opposite thing, didn't catch that. It's probably fine then, although I do think it would be better if it was more strictly typed